.github/workflows/kind-cluster-image-policy.yaml

```diff
@@ -149,7 +149,49 @@ jobs:
 
         echo '::group:: test job rejection'
         # We did not sign this, should fail
-        if kubectl create -n demo job demo2 --image=${{ env.demoimage2 }} ; then
+        if kubectl create -n demo-keyless-signing job demo2 --image=${{ env.demoimage2 }} ; then
+          echo Failed to block unsigned Job creation!
+          exit 1
+        else
+          echo Successfully blocked Job creation with unsigned image
+        fi
+        echo '::endgroup::'
+
+    - name: Generate New Signing Key
+      run: |
+        COSIGN_PASSWORD="" ./cosign generate-key-pair
+
+    - name: Deploy ClusterImagePolicy With Key Signing
+      run: |
+        yq '. | .spec.authorities[0].key.data |= load_str("cosign.pub")' ./test/testdata/cosigned/e2e/cip-key.yaml | \
+          kubectl apply -f -
+
+    - name: Sign demoimage with cosign-test key
```

Comment:
Awesomesauce! Thanks for doing this. I wonder if we should / could add one more test above this before signing.
Create the CIP with the key, this means that all CIP must be met for the image to be admitted.
Check that the image is rejected. This is now a neat test because it also tests the AND condition of the CIP. It passed above, we added a CIP that should now be rejected. 
Then sign.

So, I guess right before this step, add a test that roughly makes sure that demoimage is rejected.

Something like this:
```
      - name: Verify with two CIP one not signed with public key
      run: |
        if kubectl create -n demo-key-signing job demo --image=${{ env.demoimage }} ; then
          echo Failed to block unsigned Job creation!
          exit 1
```
What do you think?

Reply:
Added! Thanks for suggestion!